core-java-modules/core-java-jndi/src/main/java/com/baeldung/jndi/ldap/connection/tool/LdapConnectionTool.java

```diff
@@ -0,0 +1,116 @@
+package com.baeldung.jndi.ldap.connection.tool;
+
+import java.util.Hashtable;
+
+import javax.naming.Context;
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.Attribute;
+import javax.naming.directory.Attributes;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+
+public class LdapConnectionTool {
+    private static boolean DEBUG_MODE = Boolean.valueOf(System.getProperty("debug.mode", "false"));
+    private static String QUERY = "query";
+
+    public static void main(String[] args) throws NamingException {
+        execute();
+    }
+
+    public static void execute() throws NamingException {
+        Hashtable<String, String> env = createEnvironmentFromProperties();
```

Comment:
I see you are using Hashtable, so do you need a thread-safe structure for any particular reason? If not, we should favor the use of the unsynchronized equivalent structure, that is, HashMap.

Reply:
It's just that `InitialDirContext` can take only a `Hashtable`, not a `Map`, so I cannot pass a `HashMap`. So despite not needing any thread safety for this particular application, we cannot use a regular `HashMap` without some conversion overhead.

Reply:
Understood, thank you for the clarification.